### PR TITLE
Fix initial state of RGBW2

### PR DESCRIPTION
**Pull request by jobl1978**
RGBW2 wasn't able to recover the light (off -> on)  if it was in momentary mode with initial state last

### CODE REVIEW

**Comment by timoschilling** [comment]
Can you revert the code order change please, the code isn't comparable in this way or split it into two commits.

**Comment by jobl1978** [comment]
> Can you revert the code order change please, the code isn't comparable in this way or split it into two commits.

Pull request is now splitted into 2 commits

File: src/shelly_hap_rgb.cpp [hunk 1]
```diff
@@ -145,6 +116,37 @@ Status RGBWLight::Init() {
...
+    handler_id_ =
+        in_->AddHandler(std::bind(&RGBWLight::InputEventHandler, this, _1, _2));
+    in_->SetInvert(cfg_->in_inverted);
+  }
+
+  bool should_restore = (cfg_->initial_state == (int) InitialState::kLast);
+  if (IsSoftReboot()) should_restore = true;
+
+  if (should_restore) {
+    UpdateOnOff(IsOn(), "init", true);
```
Comment (timoschilling) [inline src/shelly_hap_rgb.cpp:129]:
```suggestion
    UpdateOnOff(IsOn(), "init", true /* force */);
```

**Comment by rojer** [comment]
LGTM, waiting for Timo

**Comment by timoschilling** [comment]
I have no test setup for this, but LGTM

**Pull request merged**